libyul/AsmParser.cpp

```diff
@@ -701,10 +701,19 @@ void Parser::checkBreakContinuePosition(string const& _which)
 
 bool Parser::isValidNumberLiteral(string const& _literal)
 {
+	string tmp_literal;
 	try
 	{
-		// Try to convert _literal to u256.
-		[[maybe_unused]] auto tmp = u256(_literal);
+		index = 0;
+		for(int i = 0; i < _literal.size(); i++)
+		{
+			if(_literal[i] != '_')
+			{
+				tmp_literal[index] = _literal[i];
+				index++;
+			}
+		}
+		[[maybe_unused]] auto tmp = u256(tmp_literal);
```

Comment:
May prefer `str.erase(std::remove(str.begin(), str.end(), 'a'), str.end());`

Reply:
Not may, but should definitely use this.